src/c-writer.cc

```diff
@@ -1579,6 +1584,18 @@ void CWriter::BeginInstance() {
   }
 }
 
+void CWriter::WriteUndefinedSymbolDeclarationsNoSandbox() {
+  if (module_->undefined_data_symbols_.empty()) {
+    return;
+  }
+
+  Write(Newline());
+
+  for (auto& entry : module_->undefined_data_symbols_) {
+    Write("extern char ", entry.second, ";", Newline());
```

Comment:
where do we get these?

Reply:
These are already being calculated when the symbol table is read, specifically in OnDataSymbol(...).